common/v2/services/ApiService/TokenInfo/TokenInfo.ts

```diff
@@ -21,14 +22,37 @@ export default class TokenInfoService {
   }
 
   public getTokensInfo = async (contractAddresses: string[]) => {
-    const params = new URLSearchParams();
-    contractAddresses.forEach(address => params.append('contractAddress', address));
+    // Splices contract addresses for large sets of contracts to break up the calls into manageable chunks
+    const splicedContractAddressArrays = spliceContractAddresses(contractAddresses);
+    const paramsArray = splicedContractAddressArrays.map(contractAddressArray =>
+      createParams(contractAddressArray)
+    );
 
     try {
-      const response = await this.service.get('', { params });
-      return response.data;
+      return await Promise.all(
+        paramsArray.map(async (params: any) => this.service.get('', { params }))
+      ).then((returnedArray: AxiosResponse[]) =>
+        _.union(...returnedArray.map(returnObject => returnObject.data))
+      );
```

Comment:
You can increase readability by simplifying the steps and the naming:
```
  public getTokensInfo = async (contractAddresses: string[]) => {
    // Split contract addresses for large sets of contracts to break up the calls into manageable chunks
    // Explain why you choose 20
    const queryParams = chunk(20, contractAddresses).map(createParams);

    try {
      return await Promise.all(
          queryParams.map(async (params: any) => this.service.get('', { params }))
        ).then((res: AxiosResponse[]) =>
          _.union(...res.map(r => r.data))
        );
    } catch (e) {
      throw e;
    }
  };

```
- Avoiding intermediate variables increases readability.
- In TS adding the type to the variable name is redundant and can increase cognitive load.
- You're welcome to use acronyms for standard apis.